projects/storefrontlib/src/lib/store-finder/services/store-finder.service.ts

```diff
@@ -8,43 +8,57 @@ import * as fromStore from '../store';
 
 @Injectable()
 export class StoreFinderService {
+  geolocationWatchId: number;
+
   constructor(
     private store: Store<fromStore.StoresState>,
     private winRef: WindowRef
   ) {}
 
   findStores(queryText: string, useMyLocation?: boolean) {
     if (useMyLocation) {
-      this.store.dispatch(new fromStore.OnHold());
-      this.winRef.nativeWindow.navigator.geolocation.getCurrentPosition(
+      this.clearWatchGeolocationAndDispatchAction(new fromStore.OnHold());
```

Comment:
It is probably redundant to clear watch here because you are doing it a few lines below.

Reply:
It's not redundant. Consider the case where you click Use my location really quick for two times (second time happens when watch geolocation is pending). The first clearWatch on line 20 will clear the pending watch.